dsgrid/dataset/unpivoted_table.py

```diff
@@ -61,31 +64,38 @@ def process_stacked_aggregations(
                     metric_query_name = column.dimension_query_name
 
             if metric_query_name is None:
-                raise Exception(f"Bug: A metric dimension is not included in {agg}")
+                msg = f"Bug: A metric dimension is not included in {agg}"
+                raise Exception(msg)
 
             dropped_dimensions.update(set(agg.list_dropped_dimensions()))
             if not columns:
                 continue
 
             df = self.add_columns(df, columns, context, [VALUE_COLUMN])
-            group_by_cols = self._build_group_by_columns(
-                columns, context, column_to_dim_type, dim_type_to_query_name, final_metadata
-            )
+            group_by_cols = self._build_group_by_columns(columns, context, final_metadata)
             op = agg.aggregation_function
             df = df.groupBy(*group_by_cols).agg(op(VALUE_COLUMN).alias(VALUE_COLUMN))
 
-            if metric_query_name != dim_type_to_query_name[DimensionType.METRIC]:
-                dim_config = self.project_config.get_dimension(metric_query_name)
-                mapping_records = self.project_config.get_base_to_supplemental_mapping_records(
-                    metric_query_name
+            if metric_query_name not in dim_type_to_base_query_name[DimensionType.METRIC]:
+                to_dim_config = self.project_config.get_dimension_with_records(metric_query_name)
+                mapping_configs = self.project_config.list_base_to_supplemental_mapping_configs(
+                    supplemental_dimension_id=to_dim_config.model.dimension_id,
                 )
-                to_unit_records = dim_config.get_records_dataframe()
-                df = energy.convert_units_unpivoted(
+                if len(mapping_configs) > 1:
+                    msg = (
+                        "More than one base-to-supplemental mapping config for "
+                        f"{to_dim_config.model.dimension_id}"
+                    )
+                    raise NotImplementedError(msg)
```

Comment:
I haven't figured out how to handle this.

Reply:
Do we not know the base metric dimension we're mapping from/what the current metric dimension is at this point?